src/useSetupMedia.tsx

```diff
@@ -0,0 +1,348 @@
+import type { FishjamContextType, UseSetupMediaConfig, UseSetupMediaResult } from "./types";
+import { useEffect, useMemo, useRef } from "react";
+import type { MediaDeviceType, TrackType } from "./ScreenShareManager";
+import type { ClientApi, ClientEvents } from "./Client";
+import type { PeerStatus } from "./state.types";
+
+export const createUseSetupMediaHook = <PeerMetadata, TrackMetadata>(
+  useFishjamContext: () => FishjamContextType<PeerMetadata, TrackMetadata>,
+) => {
+  const isBroadcastedTrackChanged = (
+    expectedMediaDeviceType: MediaDeviceType,
+    client: ClientApi<PeerMetadata, TrackMetadata>,
+    pending: boolean,
+    mediaDeviceType: MediaDeviceType,
+  ) =>
+    client.status === "joined" && mediaDeviceType === expectedMediaDeviceType && !pending && !client.isReconnecting();
+
+  const isBroadcastedTrackStopped = (
+    expectedMediaDeviceType: MediaDeviceType,
+    expectedTrackType: TrackType,
+    status: PeerStatus,
+    event: Parameters<ClientEvents<PeerMetadata, TrackMetadata>["deviceStopped"]>[0],
+    stream: MediaStream | undefined | null,
+  ) =>
+    status === "joined" &&
+    event.mediaDeviceType === expectedMediaDeviceType &&
+    event.trackType === expectedTrackType &&
+    stream;
```

Comment:
So this checks if track is stopped? 👀  I don't see anything related to stopping here

Reply:
I think we could implement it later